templates/default/deploy-scripts/src/deployer.rs

```diff
@@ -0,0 +1,459 @@
+use crate::v2::BlockIdentifier::Best;
+use concordium_rust_sdk::{
+    common::types::TransactionTime,
+    id::types::AccountAddress,
+    smart_contracts::common::{Address, ModuleReference},
+    types::{
+        hashes::{HashBytes, TransactionMarker},
+        queries::AccountNonceResponse,
+        smart_contracts::{ContractContext, InvokeContractResult, WasmModule},
+        transactions::{
+            self,
+            send::{deploy_module, init_contract, GivenEnergy},
+            InitContractPayload, UpdateContractPayload,
+        },
+        AccountTransactionEffects, BlockItemSummary, BlockItemSummaryDetails, ContractAddress,
+        Energy, RejectReason, TransactionType, WalletAccount,
+    },
+    v2,
+};
+use std::path::Path;
+
+use crate::DeployError;
+
+/// A struct representing the deployed module on the chain.
+#[derive(Clone, Debug)]
+pub struct ModuleDeployed {
+    /// Module reference on the chain.
+    pub module_ref: ModuleReference,
+}
+
+/// A struct representing a smart contract instance on the chain.
+#[derive(Clone, Debug)]
+pub struct ContractInitialized {
+    /// Smart contract address on the chain.
+    pub contract: ContractAddress,
+}
+
+/// A struct containing connection and wallet information.
+#[derive(Debug)]
+pub struct Deployer {
+    /// The client to establish a connection to a Concordium node (V2 API).
+    pub client: v2::Client,
```

Comment:
@DOBEN I don't think this is such a good idea.

At the cost of a few clones, which are really cheap for the client since it's an Arc, we add complexity if they do something odd.

If this was some tight inner loop it might be worthwhile, but it's not. So I recommend not doing this.